01_Data/src/layers/sequelize/repository/TransactionEvent.ts

```diff
@@ -143,11 +155,74 @@ export class SequelizeTransactionEventRepository extends SequelizeRepository<Tra
     });
   }
 
+  private async calculateAndUpdateTotalKwh(transaction: Transaction): Promise<number> {
+    const totalKwh = this.getTotalKwh(transaction.meterValues ?? []);
+
+    await transaction.update(
+      {totalKwh}
+    );
+
+    return totalKwh;
+  }
+
+  private getTotalKwh(meterValues: MeterValueType[]): number {
+    const contexts: ReadingContextEnumType[] = [
+      ReadingContextEnumType.Transaction_Begin,
+      ReadingContextEnumType.Sample_Periodic,
+      ReadingContextEnumType.Transaction_End,
+    ];
+
+    let valuesMap = new Map();
+
+    meterValues
+      .filter(
+        (meterValue) =>
+          meterValue.sampledValue[0].context &&
+          contexts.indexOf(meterValue.sampledValue[0].context) !== -1,
+      )
+      .forEach((meterValue) => {
+        const sampledValues = meterValue.sampledValue as SampledValueType[];
+        const overallValue = sampledValues.find(
+          (sampledValue) =>
+            sampledValue.phase === undefined &&
```

Comment:
this is a dangerous way to compare, it could come in as null due to Sequelize's handling of absent fields
I'd stick to a truthy-falsy comparison: 
> !sampledValue.phase &&